python/lsst/ap/association/diaCalculation.py

```diff
@@ -0,0 +1,291 @@
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from collections import namedtuple
+import numpy as np
+import pandas as pd
+
+from lsst.meas.base import (
+    BasePlugin,
+    CatalogCalculationPluginConfig,
+    CatalogCalculationPlugin,
+    CatalogCalculationConfig,
+    CatalogCalculationTask,
+    PluginRegistry,
+    PluginMap)
+from lsst.meas.base.catalogCalculation import CCContext
+import lsst.pipe.base
+
+# Enforce an error for unsafe column/array value setting in pandas.
+pd.options.mode.chained_assignment = 'raise'
+
+__all__ = ("DiaObjectCalculationPlugin", "DiaObjectCalculationPluginConfig",
+           "DiaObjectCalculationTask", "DiaObjectCalculationConfig")
+
+
+class DiaObjectCalculationPluginConfig(CatalogCalculationPluginConfig):
+    """Default configuration class DIA catalog calculation plugins.
+    """
+    pass
+
+
+class DiaObjectCalculationPlugin(CatalogCalculationPlugin):
+    """Base class for DIA catalog calculation plugins.
+
+    Task follows CatalogCalculationPlugin with modifications for use in AP.
+
+    Parameters
+    ----------
+    config : `DiaObjectCalculationPlugin.ConfigClass`
+        Plugin configuration.
+    name : `str`
+        The string the plugin was registered with.
+    metadata : `lsst.daf.base.PropertySet`
+        Plugin metadata that will be attached to the output catalog
+    """
+
+    ConfigClass = DiaObjectCalculationPluginConfig  # documentation inherited
+
+    registry = PluginRegistry(DiaObjectCalculationPluginConfig)
+    """List of available plugins (`lsst.meas.base.PluginRegistry`).
+    """
+
+    """Add order after flux means and stds are calculated.
+    """
+    FLUX_MOMMENTS_CALCULATED = 5.0
+
+    plugType = 'single'
+    """Does the plugin operate on a single source or the whole catalog (`str`)?
+    If the plugin operates on a single source at a time, this should be set to
+    ``"single"``; if it expects the whoe catalog, to ``"multi"``.  If the
+    plugin is of type ``"multi"``, the `fail` method must be implemented to
+    accept the whole catalog. If the plugin is of type ``"single"``, `fail`
+    should accept a single source record.
+    """
+
+    def __init__(self, config, name, metadata):
+        BasePlugin.__init__(self, config, name)
+
+    @classmethod
+    def getExecutionOrder(cls):
```

Comment:
I could, however, I have a feeling it may be easier to leave it here just so that one doesn't have to run looking through the stack to find what the call structure is for this method.

Reply:
Please don't do this! :-)

If you're just copying in the superclass, it's better to not bother subclassing at all!